src/screens/PinScreen.js

```diff
@@ -132,50 +138,47 @@ class PinScreen extends React.Component {
    *
    * @param {String} pin Unlock PIN written by the user
    */
-  handleDataMigration = (pin) => {
-    const accessData = hathorLib.wallet.getWalletAccessData();
-
-    if (accessData !== null && accessData.xpubkey === undefined && accessData.mainKey) {
-      // Two situations are handled here:
-      // 1. From v0.12.0 to v0.13.0 of the lib, xpubkey has changed from wallet:data
-      // to wallet:accessData. So if the user is still in an app version before v0.13.0,
-      // we can't use xpubkey directly from accessData
-      //
-      // 2. When the user updated the app to the newest version directly from a version before we've
-      // executed the xpubkey migration. In that case we have deleted the user walletData before
-      // migrating the xpubkey to the accessData
-      const xpubkey = hathorLib.wallet.getXPubKeyFromXPrivKey(pin);
-      accessData.xpubkey = xpubkey;
-      hathorLib.wallet.setWalletAccessData(accessData);
+  handleDataMigration = async (pin) => {
+    const storageVersion = STORE.getStorageVersion();
+    const oldWords = STORE.getOldWalletWords(pin);
+    if (storageVersion === null && oldWords !== null) {
+      // We are migrating from an version of wallet-lib prior to 1.0.0
+      // This will generate the encrypted keys and other metadata
+      const newAccessData = walletUtils.generateAccessDataFromSeed(
+        oldWords,
+        { pin, password: pin, networkName: NETWORK },
+      );
+      STORE.saveWalletId(newAccessData);
+      // This will return a storage using a persistent LevelDBStore
+      const storage = STORE.getStorage();
+      await storage.saveAccessData(newAccessData);
+
+      // The access data is saved on the new storage, we can delete the old data.
+      // This will only delete keys with the wallet prefix, so we don't delete
+      // the biometry keys and new data.
+      await STORE.clearItems(true);
+
+      // We have finished the migration so we can set the storage version to the most recent one.
+      STORE.updateStorageVersion();
```

Comment:
Should we set the storage version outside the `if`? You would be setting it only when you need the data migration but I feel it might be an important information, so maybe we could add it when starting the wallet as well, what do you think?

Reply:
Done